lib/tools/adb-commands.js

```diff
@@ -86,11 +86,15 @@ methods.getApiLevel = async function getApiLevel () {
       const strOutput = await this.getDeviceProperty('ro.build.version.sdk');
       let apiLevel = parseInt(strOutput.trim(), 10);
 
-      // Temp workaround. Android Q beta emulators report SDK 28 when they should be 29
-      if (apiLevel === 28 && (await this.getDeviceProperty('ro.build.version.release')).toLowerCase() === 'q') {
+      // Workaround for preview/beta platform API level
+      if (apiLevel === 28 && (await this.getPlatformVersion()).toLowerCase() === 'q') {
```

Comment:
I would rather make it dynamic, for example:

if we know that api28 + platformQ = api29 then we could assume that api(28+X) + platform(Q+X) = api(28+X+1)

Reply:
`String.fromCharCode`, ah, JS has...